### PR TITLE
Change message bus unsubscribe event driver

### LINKED ISSUE #53: New browser windows are immediately unregistered from message bus
unload event in chrome is called for new windows not just on close.

### PR DESCRIPTION
**Pull request by bingenito**
Fixes #53

### CODE REVIEW

File: src/Default/default.ts [hunk 1]
```diff
@@ -208,8 +208,8 @@ export class DefaultContainer extends WebContainerBase {
         const uuid = window[DefaultContainer.windowUuidPropertyKey] = Guid.newGuid();
         windows[uuid] = window;
 
-        // Attach unload handler to window to cleanup reference on global windows object
-        window.addEventListener("unload", (event) => delete windows[uuid]);
+        // Attach beforeunload handler to window to cleanup reference on global windows object
+        window.addEventListener("beforeunload", (event) => delete windows[uuid]);
```
Comment (dovie) [inline src/Default/default.ts]: I'm not sure about now but it used to be the case that to cover some browsers, it was worth subscribing to both events.  Not sure if that's still the case, though.

Comment (bingenito) [reply src/Default/default.ts]: @dovie this works in IE 11 too.

Comment (bingenito) [reply src/Default/default.ts]: @psmulovics You got me there.

Comment (bingenito) [reply src/Default/default.ts]: Give me an event (and no jquery)

Comment (psmulovics) [reply src/Default/default.ts]:
```
<script type="text/javascript" for="window" event="onunload">
  alert("The onunload event fired for the window object.");
</script>
```
This is supposed to work in IE11 (this is what jquery does under the hood for you)

Comment (bingenito) [reply src/Default/default.ts]: That's exactly what the code was.  Chrome fires unload after load.

Comment (bingenito) [reply src/Default/default.ts]: I can attach to unload within beforeunload

**Comment by bingenito** [comment]
- Tested in IE11 and Chrome.
- Tested application canceling beforeunload to ensure unload is still unregistering the window when user confirms closing window.

**Pull request merged**